llvm/lib/AsmParser/LLParser.cpp

```diff
@@ -5740,14 +5767,15 @@ bool LLParser::convertValIDToValue(Type *Ty, ValID &ID, Value *&V,
       V = NoCFIValue::get(cast<GlobalValue>(V));
     return V == nullptr;
   case ValID::t_APSInt:
-    if (!Ty->isIntegerTy())
+    if (!Ty->isIntOrIntVectorTy())
```

Comment:
I don't really get why these changes are needed. Isn't the value parsing itself still performed with the scalar type?

Reply:
This boils down to the same reason as above.  Floating point looks to be handled a little stricter whereas integers just resize the `ID.APSIntVal`. I've no knowledge as to the reason for this so have just maintained the current approach.